x/execution/internal/keeper/keeper.go

```diff
@@ -214,6 +231,23 @@ func (k *Keeper) Create(ctx sdk.Context, msg types.MsgCreate) (*executionpb.Exec
 		return nil, sdkerrors.Wrapf(sdkerrors.ErrJSONMarshal, err.Error())
 	}
 	store.Set(exec.Hash, value)
+
+	// emit event with action proposed
+	event := sdk.NewEvent(
+		types.EventType,
+		sdk.NewAttribute(sdk.AttributeKeyAction, types.AttributeActionProposed),
+		sdk.NewAttribute(types.AttributeKeyHash, exec.Hash.String()),
+		sdk.NewAttribute(types.AttributeKeyAddress, exec.Address.String()),
+		sdk.NewAttribute(types.AttributeKeyExecutor, exec.ExecutorHash.String()),
+		sdk.NewAttribute(types.AttributeKeyInstance, exec.InstanceHash.String()),
+	)
+	if !exec.ProcessHash.IsZero() {
+		event = event.AppendAttributes(
+			sdk.NewAttribute(types.AttributeKeyExecutor, exec.ProcessHash.String()),
+		)
+	}
+	ctx.EventManager().EmitEvent(event)
```

Comment:
It might make more sense to put that before the create. I would suggest where we actually "register" the emitter https://github.com/mesg-foundation/engine/blob/ef2bac3194e0fc93a5189f6691761cb0dac68d14/x/execution/internal/keeper/keeper.go#L169
This way we will have the event proposed before the event created.

Reply:
will do